python/mxnet/gluon/loss.py

```diff
@@ -74,6 +74,26 @@ def _reshape_like(F, x, y):
     return F.reshape_like(x, y)
 
 
+def _batch_mean(F, loss, batch_axis):
+    """Return mean on the specified batch axis, not keeping the axis"""
+    if is_np_array():
+        axes = list(range(loss.ndim))
```

Comment:
Would this work for the hybridized case?

Reply:
it won't with the current hybridization approach. it will once we fully switch to deferred compute.